pkg/asset/ignition/bootstrap/baremetal/template.go

```diff
@@ -100,6 +103,12 @@ func GetTemplateData(config *baremetal.Platform, networks []types.MachineNetwork
 
 	templateData.Hosts = config.Hosts
 
+	for _, host := range config.Hosts {
+		if host.IsMaster() {
```

Comment:
Isn't the Role optional for the user to provide? I'm not aware that we default it anywhere. See https://github.com/openshift/installer/blob/master/pkg/types/baremetal/validation/platform.go#L305-L318

`*installConfig.ControlPlane.Replicas` is the canonical source of this information.

Reply:
Addressed in a follow-up #8322